### DISCUSSION TITLE
return_url parameter

### DISCUSSION THREAD

**Discussion by rezakazemi928** [ory/kratos#2055] [Q&A] [answered]
Hi.
Is there any example or documentation to use the return_url parameter?
I checked out the documentation and I did not find any clue to solve my issue.
I set the return_to_query_param: return_to on my Oathkeeper config file and I also set the   whitelisted_return_url option on my Kratos config file. Is there something that I missed here?? 
Kratos version: v0.4.6-alpha.1
Oathkeeper version: latest

According to the documentation whenever I entered the matching URL if the user is not authenticated it would be redirected to the login page and after the authentication process to the original URL

I would appreciate that if someone could help me with this issue
Thanks

**Comment by aeneasr** [comment, marked as answer]
Does this help?

https://www.ory.sh/kratos/docs/concepts/browser-redirect-flow-completion/#specify-redirection-url-during-registration

**Reply by rezakazemi928** [reply]
Actually, I have read this before but maybe I have some misunderstanding.
The first thing I would like to as is, shouldn't the return_url parameter add automatically to the URL??
As I said I have added the  whitelisted_return_url and the return_to_query_param: return_to. let me post my config file for you to check out:

oathkeeper:

```
errors:
  fallback:
  - json
  handlers:
    redirect:
      enabled: true
      # return_to_query_param: return_to
      config:
        to: https://127.0.0.1:4455/auth/login
        return_to_query_param: return_to
        when:
        - error:
          - unauthorized
          - forbidden
          request:
            header:
              accept:
              - text/html
    json:
      enabled: true
      config:
        verbose: true
      return_to_query_param: return_to
```

```
selfservice:
  default_browser_return_url: http://127.0.0.1:4455/ #! here we can change the url
  methods:
    password:
      enabled: true
  flows:
    error:
      ui_url: http://127.0.0.1:4455/error
    settings:
      ui_url: http://127.0.0.1:4455/settings
      privileged_session_max_age: 15m
    recovery:
      enabled: true
      ui_url: http://127.0.0.1:4455/recovery
    verification:
      enabled: true
      ui_url: http://127.0.0.1:4455/verify
      after:
        default_browser_return_url: http://127.0.0.1:4455/
    logout:
      after:
        default_browser_return_url: http://127.0.0.1:4455/auth/login
    login:
      ui_url: http://127.0.0.1:4455/auth/login
      lifespan: 10m
    registration:
      lifespan: 10m
      ui_url: http://127.0.0.1:4455/auth/registration
      after:
        password:
          hooks:
          - hook: session
    
  whitelisted_return_urls:
      - http://127.0.0.1:4455/
```


And I also, added another valid external URL but nothing would show in the browser URL and I still would be redirected to the original page

**Reply by rezakazemi928** [reply]
@aeneasr
I would appreciate it if you give me a hand. I do not know what to do.

**Reply by aeneasr** [reply]
Sorry, I missed the notification. Do you have a fully reproducible docker compose set up I could try to reproduce the problem with? :)

**Reply by vinckr** [reply]
Hey @rezakazemi928  , just checking in if your problem still persists. 
I will mark this as answered for now, but feel free to come back to the discussion any time.
Thanks a ton :bee:
